packages/textfield/src/mwc-textfield-base.ts

```diff
@@ -221,26 +229,25 @@ export abstract class TextFieldBase extends FormElement {
         ${this.iconTrailing ? this.renderIcon(this.iconTrailing) : ''}
         ${this.outlined ? this.renderOutlined() : this.renderLabelText()}
       </div>
-      ${
-        (this.helper || this.validationMessage || this.charCounter) ?
-            this.renderHelperText() :
-            ''}
+      ${this.renderHelperText(this.renderCharCounter())}
     `;
   }
 
   updated(changedProperties: PropertyValues) {
-    const charCounter =
-        changedProperties.get('charCounter') as boolean | undefined;
-
-    // update foundation only when charCounter goes from false to true
-    if (!charCounter && this.charCounter) {
+    const maxLength = changedProperties.get('maxLength') as number | undefined;
+
+    const mLChangedFromUndef = maxLength === -1 && this.maxLength !== -1;
```

Comment:
the `mLC` prefix looks kinda weird. How about `maxLengthBecameDefined` and `maxLengthBecameUndefined`?

Reply:
:+1: I was initially trying to not use the term "undefined" because of the overload